rocAL/rocAL/include/meta_data.h

```diff
@@ -181,7 +234,55 @@ struct BoundingBoxBatch: public MetaDataBatch
         return std::make_shared<BoundingBoxBatch>(*this);
     }
 };
+
+struct KeyPointBatch : public MetaDataBatch
+{
+    void clear() override
+    {
+        _img_sizes.clear();
+        _joints_data = {};
+        _bb_cords.clear();
+        _bb_label_ids.clear();
+    }
+    MetaDataBatch&  operator += (MetaDataBatch& other) override
+    {
+        _img_sizes.insert(_img_sizes.end(),other.get_img_sizes_batch().begin(), other.get_img_sizes_batch().end());
+        _joints_data.image_id_batch.insert(_joints_data.image_id_batch.end(),other.get_joints_data_batch().image_id_batch.begin(),other.get_joints_data_batch().image_id_batch.end());
+        _joints_data.annotation_id_batch.insert(_joints_data.annotation_id_batch.end(),other.get_joints_data_batch().annotation_id_batch.begin(),other.get_joints_data_batch().annotation_id_batch.end());
+        _joints_data.center_batch.insert(_joints_data.center_batch.end(),other.get_joints_data_batch().center_batch.begin(),other.get_joints_data_batch().center_batch.end());
+        _joints_data.scale_batch.insert(_joints_data.scale_batch.end(),other.get_joints_data_batch().scale_batch.begin(),other.get_joints_data_batch().scale_batch.end());
+        _joints_data.joints_batch.insert(_joints_data.joints_batch.end(),other.get_joints_data_batch().joints_batch.begin(),other.get_joints_data_batch().joints_batch.end());
+        _joints_data.joints_visibility_batch.insert(_joints_data.joints_visibility_batch.end(),other.get_joints_data_batch().joints_visibility_batch.begin(),other.get_joints_data_batch().joints_visibility_batch.end());
+        _joints_data.score_batch.insert(_joints_data.score_batch.end(),other.get_joints_data_batch().score_batch.begin(),other.get_joints_data_batch().score_batch.end());
+        _joints_data.rotation_batch.insert(_joints_data.rotation_batch.end(),other.get_joints_data_batch().rotation_batch.begin(),other.get_joints_data_batch().rotation_batch.end());
+        return *this;
+    }
+    void resize(int batch_size) override
+    {
+        _joints_data.image_id_batch.resize(batch_size);
```

Comment:
please see the comment for definition. Please have a batch of jointsdata instead of defining each member inside as separate batch.

Reply:
HRNet training expects each member of JointsDataBatch as a vector instead of a vector of JointsData structure due to which we have arrived at the above format